src/main/java/com/booking/config/DatabaseConfig.java

```diff
@@ -0,0 +1,24 @@
+package com.booking.superride.config;
+
+import org.hibernate.internal.TransactionManagement;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.support.TransactionTemplate;
+
+@Configuration
+public class DatabaseConfig {
+
+    @Bean
+    TransactionTemplate readTransactionTemplate(PlatformTransactionManager platformTransactionManager){
+        var readTransactionTemplate = new TransactionTemplate(platformTransactionManager);
+        readTransactionTemplate.setReadOnly(true);
+        return readTransactionTemplate;
+    }
+
+    @Bean
+    TransactionTemplate writeTransactionTemplate(PlatformTransactionManager platformTransactionManager){

```

Comment:
Spring boot will automatically create this, so you can directly use in Service class.